qiskit/circuit/classical/expr/__init__.py

```diff
@@ -0,0 +1,202 @@
+
+Construction
+============
+
+Constructing the tree representation directly is verbose and easy to make a mistake with the
+typing.  In many cases, much of the typing can be inferred, scalar values can automatically
+be promoted to :class:`Value` instances, and any required promotions can be resolved into
+suitable :class:`Cast` nodes.
+
+The functions and methods described in this section are a more user-friendly way to build the
+expression tree, while staying close to the internal representation.  All these functions will
+automatically lift valid Python scalar values into corresponding :class:`Var` or :class:`Value`
+objects, and will resolve any required implicit casts on your behalf.
+
+.. autofunction:: lift
+
+You can manually specify casts in cases where the cast is allowed in explicit form, but may be
+losslses (such as the cast of a higher precision :class:`~.types.Uint` to a lower precision one).
+
+.. autofunction:: cast
+
+There are helper constructor functions for each of the unary operations.
+
+.. autofunction:: bit_not
+.. autofunction:: logic_not
+
+Similarly, the binary operations and relations have helper functions defined.
+
+.. autofunction:: bit_and
+.. autofunction:: bit_or
+.. autofunction:: logic_and
+.. autofunction:: logic_or
+.. autofunction:: equal
+.. autofunction:: not_equal
+.. autofunction:: less
+.. autofunction:: less_equal
+.. autofunction:: greater
+.. autofunction:: greater_equal
+
+Qiskit's legacy method for specifying equality conditions for use in conditionals is to use a
+two-tuple of a :class:`.Clbit` or :class:`.ClassicalRegister` and an integer.  This represents an
+exact equality condition, and there are no ways to specify any other relations.  The helper function
+:func:`lift_legacy_condition` converts this legacy format into the new expression syntax.
+
+.. autofunction:: lift_legacy_condition
+
+Working with the expression tree
+================================
+
+A typical consumer of the expression tree wants to recursively walk through the tree, potentially
+statefully, acting on each node differently depending on its type.  This is naturally a
+double-dispatch problem; the logic of 'what is to be done' is likely stateful and users should be
+free to define their own operations, yet each node defines 'what is being acted on'.  We enable this
+double dispatch by providing a base visitor class for the expression tree.
+
+.. autoclass:: ExprVisitor
+    :members:
+    :undoc-members:
+
+Consumers of the expression tree should subclass the visitor, and override the ``visit_*`` methods
+that they wish to handle.  Any non-overridden methods will call :meth:`~ExprVisitor.visit_generic`,
+which unless overridden will raise a ``RuntimeError`` to ensure that you are aware if new nodes
+have been added to the expression tree that you are not yet handling.
+
+For the convenience of simple visitors that only need to inspect the variabels in an expression and
```

Comment:
```suggestion
For the convenience of simple visitors that only need to inspect the variables in an expression and
```

Reply:
Thanks - I fixed this in the PR that actually introduced it.